### PR TITLE
feat(event): event detail after rsvp

### PR DESCRIPTION
**Pull request by mthinh**
# What ?
- Implement event detail screen after rsvp the event

# Screenshot 
Screenshot 2023-09-11 at 14 26 37
Screenshot 2023-09-11 at 14 26 41
Screenshot 2023-09-11 at 14 26 47

### CODE REVIEW

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                      SliverPadding(
+                        padding: EdgeInsets.only(top: Spacing.smMedium * 2),
+                      ),
+                      SliverPadding(
+                          padding: EdgeInsets.symmetric(
+                              horizontal: Spacing.smMedium),
+                          sliver: SliverToBoxAdapter(
+                            child: GuestEventDetailDashboard(event: event),
+                          )),
+                      SliverPadding(
```
Comment (roncoderpush) [inline lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart:86]: Put a `,` here

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final t = Translations.of(context);
+    return Container(
+      decoration: BoxDecoration(
+        // color: Colors.red,
+        borderRadius: BorderRadius.circular(15.r),
```
Comment (roncoderpush) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart:43]: Remove uncomment code

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                      ImagePlaceholder.defaultPlaceholder(),
+                ),
+              ),
+            ),
+          ),
+          Positioned.fill(
+              child: Container(
+            color: Colors.black.withOpacity(0.7),
+          )),
+          Padding(
```
Comment (roncoderpush) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart:70]: Add `,` and replace Container with DecoratedBox

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart [hunk 4]
```diff
@@ -29,77 +29,96 @@ class GuestEventLocation extends StatelessWidget {
...
-            child: ClipRRect(
-                borderRadius: BorderRadius.only(
-                  topLeft: Radius.circular(15.sp),
-                  bottomLeft: Radius.circular(15.sp),
+    return FutureBuilder(
+      future: isAttending
+          ? MapUtils.getLocationName(
+              lat: event.latitude ?? 0, lng: event.longitude ?? 0)
+          : Future.delayed(const Duration(milliseconds: 300), () => ''),
+      builder: (context, snapshot) => SizedBox(
```
Comment (roncoderpush) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart:37]: Why do we need an empty future method here?

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart [hunk 5]
```diff
@@ -29,77 +29,96 @@ class GuestEventLocation extends StatelessWidget {
...
+                      Align(
+                        alignment: Alignment.center,
+                        child: RippleMarker(
+                          size: 90.w,
+                          color: LemonColor.rippleMarkerColor,
+                        ),
+                      )
+                    ],
+                  )),
+            ),
```
Comment (roncoderpush) [inline lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart:71]: Add `,`

File: lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart [hunk 6]
```diff
@@ -18,8 +17,6 @@ class HomeNewsfeedListView extends StatelessWidget {
     final t = Translations.of(context);
     final colorScheme = Theme.of(context).colorScheme;
     final bloc = context.read<NewsfeedListingBloc>();
-    final refreshController = RefreshController();
-
     return BlocConsumer<NewsfeedListingBloc, NewsfeedListingState>(
```
Comment (roncoderpush) [inline lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart:20]: I tried put refreshController into BloC. It causing error that print out refreshController is using multiple times. Please put it back into UI

File: lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart [hunk 7]
```diff
@@ -39,34 +36,30 @@ class HomeNewsfeedListView extends StatelessWidget {
...
       builder: (context, state) {
         if (state.posts.isEmpty) {
-          if (state.status != NewsfeedStatus.fetched) {
-            return Loading.defaultLoading(context);
-          } else {
-            return Center(
-              child: EmptyList(emptyText: t.notification.emptyNotifications),
-            );
-          }
+          return Center(
```
Comment (roncoderpush) [inline lib/core/presentation/pages/home/views/list/home_newsfeed_list.dart:39]: Please merge latest code on master, don't remove it here

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  double get minExtent => 60;
+
+  @override
+  bool shouldRebuild(covariant SliverPersistentHeaderDelegate oldDelegate) =>
+      true;
+
+  @override
+  Widget build(
+      BuildContext context, double shrinkOffset, bool overlapsContent) {
+    final primary = Theme.of(context).colorScheme.primary;
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:40]: put `,` at the end of `overlapsContent`, make it cleaner code

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+  final bool? hideLeading;
+
+  Size get preferredSize => Size.fromHeight(60.w);
+
+  @override
+  double get maxExtent => 60;
+
+  @override
+  double get minExtent => 60;
+
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:32]: add `.w` to make it responsive

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+  final List<Widget>? actions;
+  final Color? backgroundColor;
+  final EdgeInsets? padding;
+  final bool? hideLeading;
+
+  Size get preferredSize => Size.fromHeight(60.w);
+
+  @override
+  double get maxExtent => 60;
+
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:29]: add `.w` to make it responsive

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  Widget buildCenteredTitle(String title, double shrinkOffset) {
+    return Transform.translate(
+      offset: Offset(0, maxExtent - shrinkOffset),
+      child: Text(
+        title,
+        style: const TextStyle(
+          fontWeight: FontWeight.w600,
+          fontSize: 18,
+        ),
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:65]: Use Typo

File: lib/core/presentation/widgets/event/event_dashboard_item.dart [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+              borderRadius: BorderRadius.circular(16),
+            ),
+          ),
+          child: Row(
+            mainAxisSize: MainAxisSize.max,
+            children: [
+              Expanded(
+                child: Container(
+                  height: 111,
+                  padding: const EdgeInsets.all(15),
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/event/event_dashboard_item.dart:43]: Add `.h`

File: lib/core/presentation/widgets/event/event_dashboard_item.dart [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+                  clipBehavior: Clip.antiAlias,
+                  decoration: ShapeDecoration(
+                    gradient: LinearGradient(
+                      begin: const Alignment(0.00, -1.00),
+                      end: const Alignment(0, 1),
+                      colors: [LemonColor.chineseBlack, Colors.black],
+                    ),
+                    shape: RoundedRectangleBorder(
+                      borderRadius: BorderRadius.circular(15),
+                    ),
```
Comment (roncoderpush) [inline lib/core/presentation/widgets/event/event_dashboard_item.dart:53]: add `.r`

**Review by roncoderpush — CHANGES_REQUESTED** [review]

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+        builder: (context, state) => state.when(
+          failure: () => Center(
+              child: EmptyList(
+            emptyText: t.common.somethingWrong,
+          )),
+          loading: () => Loading.defaultLoading(context),
+          fetched: (event) {
+            return SafeArea(
+              child: Stack(
+                children: [
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart:41]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart (hunk 1 above)
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart:86]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+                      ),
+                      if (event.latitude != null &&
+                          event.longitude != null) ...[
+                        SliverPadding(
+                          padding: EdgeInsets.only(top: Spacing.smMedium),
+                        ),
+                        SliverPadding(
+                          padding: EdgeInsets.symmetric(
+                              horizontal: Spacing.smMedium),
+                          sliver: SliverToBoxAdapter(
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/views/post_guest_event_detail_view.dart:97]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/views/pre_guest_event_detail_view.dart [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+  const PreGuestEventDetailView({super.key});
+
+  @override
+  State<PreGuestEventDetailView> createState() =>
+      PreGuestEventDetailViewState();
+}
+
+class PreGuestEventDetailViewState extends State<PreGuestEventDetailView> {
+  late final ScrollController _scrollController = ScrollController();
+  bool _isSliverAppBarCollapsed = false;
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/views/pre_guest_event_detail_view.dart:33]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_about_card.dart [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+                imageUrl: photoUrl,
+                errorWidget: (_, __, ___) => ImagePlaceholder.eventCard(),
+                placeholder: (_, __) => ImagePlaceholder.eventCard(),
+              ),
+            ),
+          ),
+          Positioned.fill(
+            child: Container(
+              decoration: BoxDecoration(
+                borderRadius: BorderRadius.circular(15.r),
```
[the remark this answers is not in the document]
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_about_card.dart:60]: 👍

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart (hunk 10 above)
Comment (mthinh) [reply lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:29]: 👍

File: lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart (hunk 11 above)
Comment (mthinh) [reply lib/core/presentation/widgets/common/appbar/lemon_animated_appbar_widget.dart:65]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart (hunk 2 above)
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_detail_clock.dart:43]: 👍

File: lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart (hunk 4 above)
Comment (mthinh) [reply lib/core/presentation/pages/event/guest_event_detail_page/widgets/guest_event_location.dart:37]: I dont want to call gg service if user not attend the event yet, so jsut a fake future , the value of emptry will not be used

**Review by roncoderpush — APPROVED** [review]

**Review by sieuhuflit — APPROVED** [review]

**Pull request merged**